Month3/LinkedList/LinkedList/TSLList.h

```diff
@@ -0,0 +1,334 @@
+#pragma once
+# include <iostream>
+#include <string>
+#include <functional>
+template <typename T>
+class TSLList
+{
+private:
+	
+	struct FNode
+	{
+		T Element;
+		FNode* Next = nullptr ;
+	};
+	
+	FNode* Head = nullptr;
+	FNode* Tail = nullptr;
+	
+	int Size = 0;
+public:
+	TSLList()
+	{
+
+	}
+	TSLList(const T& NewElement)
+	{
+		Head = new FNode();
+		Head->Element = NewElement;
+		Tail = Head;
+		Size = 1;
+	}
+
+	TSLList(const TSLList<T>& OtherList)
+	{
+		FNode* Current = OtherList.Head;
+		for (int i = 0; i < OtherList.Size; i++)
+		{
+			AddTail(Current->Element);
+			Current = Current->Next;
+		}
+	}
+
+	TSLList& operator=(const TSLList<T>& OtherList)
+	{
+		Clear();
+		FNode* Current = OtherList.Head;
+		for (int CurrentIndex = 0; CurrentIndex < OtherList.Size; CurrentIndex++)
+		{
+			AddTail(Current->Element);
+			Current = Current->Next;
+		}
+		return *this;
+	}
+
+	~TSLList()
+	{
+		Clear();
+	}
+
+
+
+	//Access
+	
+
+	T& operator[](const int Index) 
+	{
+		FNode* Current = Head;
+		for (int CurrentIndex = 0; CurrentIndex < Index; CurrentIndex++) 
+		{
+			Current = Current->Next;
+		}
+		
+		return Current->Next;
+	}
+	const T& operator[](const int Index) const
+	{
+		FNode* Current = Head;
+		for (int CurrentIndex = 0; CurrentIndex < Index; CurrentIndex++)
+		{
+			Current = Current->Next;
+		}
+
+		return Current->Next;
+	}
+
+
+	T& GetHead() 
+	{
+		return Head->Element;
+	}
+	const T& GetHead() const
+	{
+		return Head->Element;
+	}
+
+	T& GetTail() 
+	{
+		return Tail->Element;
+	}
+	const T& GetTail() const
+	{
+		return Tail->Element;
+	}
+	
+
+
+
+	//Size-------------------------------------
+
+	bool IsEmpty()
+	{
+		return Size == 0;
+	}
+
+	int GetSize() 
+	{
+		return Size;
+	}
+
+
+	// Modificators
+
+	void AddHead(const T& NewElement) 
+	{
+		
+		FNode* NewNode = new FNode;
+		NewNode->Element = NewElement;
+		NewNode->Next = Head;
+		Head = NewNode;
+		if (Size == 0) 
+		{
+			Tail = NewNode;
+		}
+		
+		Size++;
+	}
+
+	void AddTail(const T& NewElement) 
+	{
+		if (Tail)
+		{
+			FNode* NewNode = new FNode;
+			NewNode->Element = NewElement;
+			NewNode->Next = nullptr;
+
+			Tail->Next = NewNode;
+			Tail = NewNode;
+
+			Size++;
+		}
+		else 
+		{
+			AddHead(NewElement);
+		}
+	}
+
+	void Insert(const T& NewElement, const int Index )
+	{
+		if (Index < 0 || Index > Size) 
+		{
+			return;
+		}
```

Comment:
Great, but I think it's easier to understand if you do the correct case :)
what I mean is:
```
if(Index > 0 && Index < Size)
{...
 ```

Reply:
David, said in my code review that this way it's easier to understand xd, now is in your hands xd

Reply:
in my code I validate first and after this I make dthe function, in this context is understandable